### DISCUSSION TITLE
bg-* styles don't work when using @apply on button elements

### DISCUSSION THREAD

**Discussion by alkavan** [tailwindlabs/tailwindcss#7965] [Help]
**What version of Tailwind CSS are you using?**

`tailwindcss v3.0.23`

**What build tool (or framework if it abstracts the build tool) are you using?**

`postcss-cli v8.4.12`, `autoprefixer v10.4.4`

**What version of Node.js are you using?**

`node v14.19.0`

**What browser are you using?**

Chrome, Firefox

**What operating system are you using?**

Linux

**Describe your issue**

So here are two examples that I tried to make my button have blue background by default.

Using `@layer base {}`:
```
@layer base {
    button {
        @apply px-3 py-1 text-white border-2 border-solid border-blue-200 bg-blue-700;
    }
}
```

And using a normal class definition:
```
.theme-button {
    @apply px-3 py-1 text-white border-2 border-solid border-blue-200 bg-blue-700;
}
```

In both cases, the result is all styles are applied to button, however the background (`bg-*`) is not.

It's also worth noting if adding the `bg-blue-700` class directly to the element it does work, 
and also applying the class to body seems to work:

```
body {
    @apply bg-blue-700;
}
```

Also, this happened in the context of a simple React application, not sure if that should matter - but perhaps related.

**Comment by adamwathan** [comment]
Hey! This definitely works:

https://play.tailwindcss.com/OBmsifpHfe

You’ll have to include a reproduction for anyone to be able to help any further.

**Reply by alkavan** [reply]
I see.
Okay I'll try adding reproduction of this!
This maybe indeed related to React somehow.

**Reply by enzon19** [reply]
I have the same problem with Svelte. But for hover:bg-* works... I really don't know what's going on.
